pkg/stacks/pod_test.go

```diff
@@ -183,57 +183,93 @@ func TestGetSpecContainerImage(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := GetSpecContainerImage(tt.pod.Spec, tt.containerName, tt.initContainer)
+			got, err := GetContainerImage(tt.pod, tt.containerName, tt.initContainer)
 
 			if diff := cmp.Diff(err, tt.want.err, test.EquateErrors()); diff != "" {
-				t.Errorf("GetSpecContainerImage() want error != got error:\n%s", diff)
+				t.Errorf("GetContainerImage() want error != got error:\n%s", diff)
 			}
 
 			if diff := cmp.Diff(got, tt.want.image); diff != "" {
-				t.Errorf("GetSpecContainerImage() got != want:\n%v", diff)
+				t.Errorf("GetContainerImage() got != want:\n%v", diff)
 			}
 		})
 	}
 }
 
-func TestGetContainerImage(t *testing.T) {
+func TestGetContainerImagePullPolicy(t *testing.T) {
```

Comment:
I feel like we may want a test for when the image is valid and no pull policy is specified. I don't think I see one